lib/ecto/changeset.ex

```diff
@@ -729,6 +738,9 @@ defmodule Ecto.Changeset do
         case Map.get(types, key) do
           {:embed, embed} ->
             {key, Ecto.Embedded.apply_changes(embed, value)}
+          {:assoc, _assoc} ->
+            raise ArgumentError, "associations should be removed from changeset" <>
```

Comment:
@sumerman are you advocating for `apply_changes` to also apply the association changes?